lib/heidrun/mapping_tools/marc/dctype.rb

```diff
@@ -0,0 +1,125 @@
+module Heidrun
+  module MappingTools
+    module MARC
+      ##
+      # Methods that are used in assigning dctype values
+      #
+      module DCType
```

Comment:
Could we avoid the side effects in this module? E.g., something like:

```ruby
##
# @return [Array] the 337a fields; empty if none are found.
def get_337a(opts = {})
  opts.fetch(:df_337a, [])
end
```

Then the caller can do:

```ruby
def dctype(opts)
  types = []
  types.concat(DCType.get_337a(opts))
  ...
  types
end
```

It seems like this would result in a lot less shared edifice between the modules, and reduces the test scenario for the smaller methods to `expect(get_337a({:df_337a => value})).to eq value` and `expect(get_337a({})).to be_empty`.